orchagent/pfc_detect_mlnx.lua

```diff
@@ -0,0 +1,86 @@
+-- KEYS - queue IDs
+-- ARGV[1] - counters db index
+-- ARGV[2] - counters table name
+-- ARGV[3] - poll time interval
+-- return queue Ids that satisfy criteria
+
+local counters_db = ARGV[1]
+local counters_table_name = ARGV[2]
+local poll_time = tonumber(ARGV[3])
+
+local rets = {}
+
+redis.call('SELECT', counters_db)
+
+-- Iterate through each queue
+local n = table.getn(KEYS)
+for i = n, 1, -1 do
+    local counter_keys = redis.call('HKEYS', counters_table_name .. ':' .. KEYS[i])
+    local counter_num = 0
+    local old_counter_num = 0
+    local is_deadlock = false
+    local pfc_wd_status = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_STATUS')
+    if pfc_wd_status == 'operational' then
```

Comment:
why do we have two lua script? It looks like both script iterate through the queue, depending on the pfc_wd_status (operational or not), it does either detection or check if storm is stopped. Why not just have one lua script?

Reply:
You are correct. It's left from the previous implementation.